### PR TITLE
Bug 1989610: Don't render empty div for invalid descriptors.

### PR DESCRIPTION
**Pull request by TheRealJon**
- Move div wrapper for descriptor details items down to SpecDescriptorDetailsItem and StatusDescriptorDetailsItem components so that the div itself is also conditionally rendered based on descriptor validation.
- Move duplicate Secret components out of status and spec descriptor index files and into the common component file.

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/index.tsx [hunk 1]
```diff
@@ -316,49 +282,60 @@ const UpdateStrategy: React.FC<SpecCapabilityProps> = ({
...
+          console.warn(
+            `[Invalid SpecDescriptor] Cannot render 'spec.${props.descriptor.path}'. A valid x-descriptor must be provided for non-primitive properties.`,
+            'See https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#olm-descriptor-reference',
+          );
+          return null;
+        }
+        return DefaultCapability;
+    }
+  }, [props.descriptor, props.value, capability]);
+  return !Component ? null : (
```
Comment (andrewballantyne) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/index.tsx:334]: Not a big deal, but you could write this as 
```suggestion
  return Component && (
```

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Comment by TheRealJon** [comment]
@andrewballantyne I've made some updates to address your comments

**Review by andrewballantyne — APPROVED** [review]
/lgtm

**Pull request merged**